### ISSUE TITLE
Insufficient Buffer Error when Re-deriving SRK

### ISSUE THREAD

**Issue by bit-bender** [NationalSecurityAgency/kmyth#8]
When running kmyth-seal, the following error message is being observed:

kmyth-0.0.0 ERROR - src/util/tpm2_kmyth_object.c(tpm2_kmyth_create_object:353) Tss2_Sys_CreatePrimary(): rc = 0x00090006, mu:A buffer isn't large enough ... exiting

Enabling more detailed logging in TSS reveals the underlying marshaling error:

debug:marshal:src/tss2-mu/tpm2b-types.c:302:Tss2_MU_TPM2B_DIGEST_Marshal() buffer_size: 4096 with offset: 30 are insufficient for object of size 32769

**Comment by b-carr** [comment]
I was seeing this issue in my environment but am not anymore. If you re-run the command do you get different values for the object size?

**Comment by bit-bender** [comment]
In my setup, I was pretty consistently getting an insufficient buffer error where a TPM2B_DIGEST struct (on the order of 32K bytes) was attempted to be marshaled into a command buffer with a size of 4096 bytes, which was consistent with the TPM2_PT_MAX_COMMAND_SIZE setting of 0x1000 in my simulator. After finding the issue (see below), I'm not sure why it was not more sporadic (or at least showing more variation in the object size values), as you experienced.

When I traced through the code, I found that for the authPolicy digest (a TPM2B_DIGEST struct) in the object template, the size field was only being set (as part of the template setup) when the authPolicy digest was non-empty. For the empty digest case (as when re-deriving the SRK), the size field was not being set, so it would be corrupted if non-zeroed memory existed at this location. To correct this issue, I moved the size assignment outside of an if/else conditional, so that it is always set to something (e.g., zero if an empty authPolicy digest is specified).
